final-project/static/scripts.js

```diff
@@ -528,6 +529,69 @@ function insertSaveButton() {
 
 }
 
+// Provide element(s) with delete functionality.
+
+// Append a delete button to the elements.
+function insertDeleteButton(appendToTheseElements) {
+
+  // Return a new promise when this resolves.
+  return new Promise((resolve,reject) => {
+
+  if (appendToTheseElements instanceof NodeList) {
+
+    // Go through each element in a node list.
+    appendToTheseElements.forEach(
+      function(currentValue, currentIndex, listObj) {
+        // Create a delete button.
+        var deleteButton = document.createElement('input');
+        deleteButton.setAttribute('value', 'x');
+        deleteButton.setAttribute('type', 'button');
+        deleteButton.setAttribute('class', 'delete-element');
+
+        if (appendToTheseElements.id != null) {
+          deleteButton.setAttribute('id', 'delete-btn-for' + appendToTheseElements.id);
+        }
+
+        // TODO: Append the delete button to the anchor tag
+
+        // Append the delete button to the element.
+        this.appendChild(deleteButton);
+
+        // Log information in the console.
+        console.log(currentValue + ', ' + currentIndex + ', ' + this);
+      },
+      'myThisArg'
+    );
+  } else {
+    // Create a delete button.
+    var deleteButton = document.createElement('input');
+    deleteButton.setAttribute('value', 'x');
+    deleteButton.setAttribute('type', 'button');
+    deleteButton.setAttribute('class', 'delete-element');
+
+    if (appendToTheseElements.id != null) {
+      deleteButton.setAttribute('id', appendToTheseElements.id + '-delete-btn');
+    }
+
+    // Append the delete button to the element.
+    appendToTheseElements.insertAdjacentElement('afterend', deleteButton);
+  }
+
+  // Resolve the promise.
+  resolve("Delete buttons have been created");
+  });
+}
+
+// TODO Listen for clicks on the delete button.
+function listenForDelete() {
+  var deleteButtons = querySelectorAll('.delete-element');
+}
+
+// TODO Delete data from the database.
+function deleteFromDatabase(dataToDelete) {
```

Comment:
I hope this won't be writing directly to DB. It's not necessary for the client to understand the server-side concepts of data. A `delete` would suffice.